openedx/core/djangoapps/user_authn/views/tests/test_register.py

```diff
@@ -1600,15 +1601,27 @@ def _assert_fields_match(self, actual_field, expected_field):
             msg=u"Could not find field {name}".format(name=expected_field["name"])
         )
 
-        for key in expected_field:
-            self.assertEqual(
-                actual_field[key], expected_field[key],
-                msg=u"Expected {expected} for {key} but got {actual} instead".format(
-                    key=key,
-                    actual=actual_field[key],
-                    expected=expected_field[key]
+        actual_options = actual_field.get("options")
+        expected_options = expected_field.get("options")
+
+        if actual_options and expected_options:
```

Comment:
why adding if else in test ?

Reply:
I think we might want to change our implementations instead of changing tests. To make sure it's working the same.